geomstats/special_orthogonal_group.py

```diff
@@ -215,13 +227,55 @@ def jacobian_translation(self, rot_vec,
             coef_2 = (1 - coef_1) / angle ** 2
 
         if left_or_right == 'left':
-            jacobian = (coef_1 * np.identity(3)
-                        + coef_2 * np.outer(rot_vec, rot_vec)
-                        + skew_matrix_from_vector(rot_vec) / 2)
+            jacobian = (coef_1 * np.identity(self.dimension)
+                        + coef_2 * np.outer(point, point)
+                        + skew_matrix_from_vector(point) / 2)
 
         else:
-            jacobian = (coef_1 * np.identity(3)
-                        + coef_2 * np.outer(rot_vec, rot_vec)
-                        - skew_matrix_from_vector(rot_vec) / 2)
+            jacobian = (coef_1 * np.identity(self.dimension)
+                        + coef_2 * np.outer(point, point)
+                        - skew_matrix_from_vector(point) / 2)
 
         return jacobian
+
+    def random_uniform(self):
+        """
+        Sample a 3d rotation vector uniformly, w.r.t.
+        the bi-invariant metric, by sampling in the
+        hypercube of side [-1, 1] on the tangent space.
+        """
+        random_rot_vec = np.random.rand(self.dimension) * 2 - 1
+        random_rot_vec = self.regularize(random_rot_vec)
+        return random_rot_vec
+
+    def group_exp_from_identity(self, tangent_vec):
+        """
+        Compute the group exponential of vector tangent_vector.
+        """
+        tangent_vec = self.regularize(tangent_vec)
+        return tangent_vec
+
+    def group_log_from_identity(self, point):
+        """
+        Compute the group logarithm of point point.
+        """
+        point = self.regularize(point)
+        return point
+
+    def group_exponential_barycenter(self, points, weights=None):
+        """
+        Group exponential barycenter is the Frechet mean
+        of the bi-invariant metric.
```

Comment:
That's one of the most badass comments I've read. Ever.

Btw if you want to loose 1/2 hour having fun:

https://stackoverflow.com/questions/184618/what-is-the-best-comment-in-source-code-you-have-ever-encountered

Here, I've just destroyed your productivity for a nice chunk of the day. You're welcome.